cibyl/sources/jenkins.py

```diff
@@ -356,28 +356,36 @@ def get_tests(self, **kwargs):
                     build_id=build['number'],
                     status=build['result'])
 
+                job_objects[job_name].add_build(build_object)
+
                 if build['result'] == 'FAILURE':
                     LOG.warning("Build %s for job %s failed. No tests to "
                                 "fetch", build['number'], job_name)
                     continue
 
                 # Get the tests for this build
-                tests_found = self.send_request(
-                    item=f"job/{job_name}/{build['number']}/testReport",
-                    query=self.build_tests_query)
+                try:
+                    tests_found = self.send_request(
+                        item=f"job/{job_name}/{build['number']}/testReport",
+                        query=self.build_tests_query)
+                except JenkinsError as jerr:
+                    if re.search('404', str(jerr)):
```

Comment:
You can check if there is a string in another (you have some examples in different answers here: https://stackoverflow.com/a/3437070/12771230 ). Maybe it's better than using regex for this.

Reply:
That's better, I've just changed that. Thanks a lot!